sdk-lib/src/main/java/cash/z/ecc/android/sdk/model/TransactionOverview.kt

```diff
@@ -21,7 +26,62 @@ data class TransactionOverview internal constructor(
     val receivedNoteCount: Int,
     val sentNoteCount: Int,
     val memoCount: Int,
-    val blockTimeEpochSeconds: Long
+    val blockTimeEpochSeconds: Long,
+    val transactionState: TransactionState
 ) {
     override fun toString() = "TransactionOverview"
+
+    companion object {
+        internal fun new(
+            dbTransactionOverview: DbTransactionOverview,
+            latestBlockHeight: BlockHeight
+        ): TransactionOverview {
+            return TransactionOverview(
+                dbTransactionOverview.id,
+                dbTransactionOverview.rawId,
+                dbTransactionOverview.minedHeight,
+                dbTransactionOverview.expiryHeight,
+                dbTransactionOverview.index,
+                dbTransactionOverview.raw,
+                dbTransactionOverview.isSentTransaction,
+                dbTransactionOverview.netValue,
+                dbTransactionOverview.feePaid,
+                dbTransactionOverview.isChange,
+                dbTransactionOverview.isWalletInternal,
+                dbTransactionOverview.receivedNoteCount,
+                dbTransactionOverview.sentNoteCount,
+                dbTransactionOverview.memoCount,
+                dbTransactionOverview.blockTimeEpochSeconds,
+                TransactionState.new(
+                    latestBlockHeight,
+                    dbTransactionOverview.minedHeight,
+                    dbTransactionOverview.expiryHeight
+                )
+            )
+        }
+    }
+}
+
+enum class TransactionState {
+    Confirmed,
+    Pending,
+    Expired;
+
+    companion object {
+
+        private const val MIN_CONFIRMATIONS = 10
```

Comment:
I think this is already a constant called defaultStaleTolerance or so.

Reply:
The only thing I can find is ZcashSdk.EXPIRY_OFFSET which is about how many blocks before a transaction expires.